jina/clients/base/http.py

```diff
@@ -77,6 +77,13 @@ def _result_handler(result):
                     r_str = await response.json()
                     if r_status == 404:
                         raise BadClient(f'no such endpoint {url}')
+                    elif (
+                        r_status == 503
+                        and 'header' in r_str
+                        and 'status' in r_str['header']
+                        and 'description' in r_str['header']['status']
```

Comment:
What happens if the response is 503 and there is no description? We probably don't do this intentionally, but I guess this could still happen if sth unexpected happened at the Gateway? Won't we hide the error then in the client?
Might be good to cover such a scenario in a test?